primitives/src/witness.rs

```diff
@@ -185,37 +185,41 @@ impl Witness {
 
     /// Returns the last element in the witness, if any.
     #[inline]
-    pub fn last(&self) -> Option<&[u8]> {
-        if self.witness_elements == 0 {
-            None
-        } else {
-            self.nth(self.witness_elements - 1)
-        }
-    }
-
-    /// Returns the second-to-last element in the witness, if any.
-    #[inline]
-    pub fn second_to_last(&self) -> Option<&[u8]> {
-        if self.witness_elements <= 1 {
-            None
-        } else {
-            self.nth(self.witness_elements - 2)
-        }
-    }
+    pub fn last(&self) -> Option<&[u8]> { self.get_back(0) }
 
-    /// Returns the third-to-last element in the witness, if any.
-    #[inline]
-    pub fn third_to_last(&self) -> Option<&[u8]> {
-        if self.witness_elements <= 2 {
+    /// Retrieves an element from the end of the witness by its reverse index.
+    ///
+    /// `index` is 0-based from the end, where 0 is the last element, 1 is the second-to-last, etc.
+    ///
+    /// Returns `None` if the requested index is beyond the witness's elements.
+    ///
+    /// # Examples
+    /// ```
+    /// use bitcoin_primitives::witness::Witness;
+    ///
+    /// let mut witness = Witness::new();
+    /// witness.push(b"A");
+    /// witness.push(b"B");
+    /// witness.push(b"C");
+    /// witness.push(b"D");
+    ///
+    /// assert_eq!(witness.get_back(0), Some(b"D".as_slice()));
+    /// assert_eq!(witness.get_back(1), Some(b"C".as_slice()));
+    /// assert_eq!(witness.get_back(2), Some(b"B".as_slice()));
+    /// assert_eq!(witness.get_back(3), Some(b"A".as_slice()));
+    /// assert_eq!(witness.get_back(4), None);
+    /// ```
+    pub fn get_back(&self, index: usize) -> Option<&[u8]> {
+        if self.witness_elements <= index {
             None
         } else {
-            self.nth(self.witness_elements - 3)
+            self.get(self.witness_elements - 1 - index)
```

Comment:
Could you just use https://doc.rust-lang.org/std/primitive.u8.html#method.wrapping_sub here?  Also see: https://users.rust-lang.org/t/idiomatic-way-to-get-second-last-third-last-fourth-last-elements-of-a-vector-array-slice/59495/4 if you haven't already.

Reply:
No, that would be worse. `wrapping` means "this can overflow and we want it to" which is not the case here. The code is correct.